### DISCUSSION TITLE
Setting for delay start when not enough grid power

### DISCUSSION THREAD

**Discussion by negronoir** [serkri/SmartEVSE-3#227] [Q&A]
Is there any setting (even in source code) to delay restart of charge when not enough grid power available?

I have checked the 'ChargeDelay' variable but I'm a bit confused if only works in solar mode.

I'd like to set a long (e.g. 10 min.) timer to restart charge if not enough power from grid.

The reason is that I charge the car until 7:00a.m. but on working days between ~ 5:45-6:30a.m. when we wake up at home, due to high power consumption of another EV charging plus coffee machine, microwave, bathroom heaters, boiler, etc. causes frequent stops and starts of my EV charge.

![image](https://github.com/serkri/SmartEVSE-3/assets/53239888/7dfda0e0-ca00-4f6a-8380-bf53934bdfb2)

Thanks!

Toni

**Comment by dingo35** [comment]
Why don't you let the Smart mode do that work for you? It starts charging as soon as there is enough power available, and stops when there's not.
If you don't like the on and off switch (why?), you can use the Delayed Charge option on the webserver, to schedule your charge to a time of day that is more suitable.

**Reply by negronoir** [reply]
I think that I have not explained well the issue

The SmartEVSE is controlled with your Home Assistant integration. Switches from OFF to SMART at 01:00 a.m. and from SMART to OFF at 07:00a.m.
During those hours the charging starts and stops that occur some days between 5:30 and 6:30 a.m. are managed by SMART mode.

The available power varies every minute between 4A and 12A which causes the SMART mode to start and stop charging many times.

What I want to minimize is that high number of starts and stops in such a short time.

That's why I had thought that:

-If the power drops below 6A, when the SMART MODE stops charging, start a timer e.g. 10 minutes.
-If before that 10 min. timer expires the power goes up from 6A and goes back down from 6A, the timer restarts
-When 10 min have passed without the available power falling below 6A, the charging restarts
